pkg/proxy/endpoints.go

```diff
@@ -104,7 +105,8 @@ type EndpointChangeTracker struct {
 	// items maps a service to is endpointsChange.
 	items map[types.NamespacedName]*endpointsChange
 	// makeEndpointInfo allows proxier to inject customized information when processing endpoint.
-	makeEndpointInfo makeEndpointFunc
+	makeEndpointInfo     makeEndpointFunc
+	onEndpointsMapChange endpointsMapChangeFunc
```

Comment:
Technically, this does not get called on endpoints map change. It is called during `apply` with a batch of changes. This function is sort of like an addon/extension to the apply function. 

Just a suggestion: name it to something like `processEndpointsMapChange`